maestro-orchestra/src/main/java/maestro/orchestra/yaml/YamlCommandReader.kt

```diff
@@ -38,25 +42,73 @@ class YamlCommandReader : CommandReader {
             }
     }
 
-    override fun readCommands(source: Source): List<MaestroCommand> {
-        val fluentCommands = try {
-            mapper.readValue(
-                source.buffer().inputStream(),
-                object : TypeReference<List<YamlFluentCommand>>() {}
-            )
+    // Files to watch for changes. Includes any referenced files.
+    fun getWatchFiles(flowFile: File): List<File> {
+        return listOfNotNull(
+            flowFile,
+            getInitFlowFile(flowFile)
+        ).filter { it.parentFile.isDirectory }
+    }
+
+    fun readCommands(flowFile: File): Pair<List<MaestroCommand>, List<MaestroCommand>> {
```

Comment:
It should be a responsibility of the caller to extract init commands. This class is only concerned with parsing commands and returning them back

Reply:
Updated